extensions/2.0/Khronos/KHR_lights_image_based/README.md

```diff
@@ -0,0 +1,225 @@
+                    [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+                ]
+            }
+        ]
+    }
+}
+```
+
+An example using a cubemap for diffuse irradiance:
+
+```json
+"extensions": {
+    "KHR_lights_image_based" : {
+        "imageBasedLights": [
+            {
+                "rotation": [0, 0, 0, 1],
+                "brightnessFactor": 1.0,
+                "brightnessOffset": 0.0,
+                "specularEnvironmentTexture": 0,
+                "diffuseEnvironmentTexture": 1
+            }
+        ]
+    }
+}
+```
+
+## Specular BRDF integration and Irradiance Coefficients
+
+This extension uses a prefiltered environment map to define the specular lighting utilizing the split sum approximation. More information:
+- [Specular BRDF integration in Google Filament](https://google.github.io/filament/Filament.md.html#lighting/imagebasedlights/processinglightprobes)
+- [Pre-Filtered Environment Map in Unreal Engine 4](https://blog.selfshadow.com/publications/s2013-shading-course/karis/s2013_pbs_epic_notes_v2.pdf)
+
+This extension can use spherical harmonic coefficients to define irradiance used for diffuse lighting. Coefficients are calculated for the first 3 SH bands (l=2) and take the form of a 9x3 array. More information:
+- [Realtime Image Based Lighting using Spherical Harmonics](https://metashapes.com/blog/realtime-image-based-lighting-using-spherical-harmonics/)
+- [An Efficient Representation for Irradiance Environment Maps](http://graphics.stanford.edu/papers/envmap/)
+
+## Adding Light Instances to Scenes
+
+Each scene can have a single IBL light attached to it by defining the `extensions` and `KHR_lights_image_based` property and, within that, an index into the `imageBasedLights` array using the `imageBasedLight` property.
+
+```json
+"scenes" : [
+    {
+        "extensions" : {
+            "KHR_lights_image_based" : {
+                "imageBasedLight" : 0
+            }
+        }
+    }            
+]
+```
+
+## Shader Code Implementation
+
+```
+finalSampledColor = sampledColor * brightnessFactor + brightnessOffset;
+```
+
+## Image Encoding
+
+### Texture Restrictions
+
+- A glTF `texture` used for IBL cannot have a `KHR_texture_transform` extension defined.
+- `specularEnvironmentTexture` and `diffuseEnvironmentTexture` (when used) must refer to a texture referrring to a KTX2 image of **Cubemap** type as defined in KTX2, Section 4.3. Namely:
+  - `pixelHeight` must be greater than 0.
+  - `pixelDepth` must be 0.
+  - `numberOfArrayElements` must be 0.
+  - `numberOfFaces` must be 6.
+- The image must contain mip levels.
+- The transfer function for all image formats must be linear.
+- Cubemaps orientation must be aligned with [glTF coordinate system](../../../../specification/2.0#coordinate-system-and-units). 
+
+### Normal Quality
+
+Data must be stored as a CTTF image with alpha channel conforming to RGBD (or RGBE, **TBD**) encoding.
```

Comment:
I'm assuming that we're suggesting CTTF here so that the texture can remain compressed in memory. However, block-based compression schemes really don't work well with RGBE because they don't take into account that the alpha is being used as an exponent. The result is usually a lot of ugly banding between exponent boundaries. I don't remember how RGBD behaves when block-based compression is used but I expect it would be better. We should do some tests.

Reply:
RGBD, RGBE or RGBM should be decoded on CPU before uploaded to GPU. So, in general, not the perfect format but defined as mostly used.

Reply:
> I'm assuming that we're suggesting CTTF here so that the texture can remain compressed in memory. 

Regardless of an effective GPU format, CTTF should give us better transmission bandwidth than typical web formats. 

> RGBD, RGBE or RGBM should be decoded on CPU before uploaded to GPU.

CPU decoding of RGBD (RGBE) from CTTF is expected to be supported.